### PR TITLE
Output pseudo-paragraph

### PR DESCRIPTION
**Pull request by maxaalexeeva**
Adding a +/-n sentence context window column/field to output. Getting to real paragraph looks like it would complicate some methods + paragraphs we get from pdfs are not always reliable and they might be too big (thus overwhelming within a document). Context window size is negotiable (set in application.conf).

### CODE REVIEW

File: src/main/scala/org/clulab/habitus/printer/MentionInfo.scala [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
+case class MentionInfo(contextWindow: String, sentenceText: String, inputFilename: String, label: String) extends Pairable
 
 object MentionInfo {
 
+  val config = ConfigFactory.load()
+  val windowSize: Int = config[Int]("VarDatesReader.contextWindowSize")
+
   def apply(mention: Mention, inputFilename: String): MentionInfo = {
     val document = mention.document
+    val sentInterval = (math.max(mention.sentence - windowSize, 0), math.min(mention.sentence + windowSize + 1, mention.document.sentences.length + 1))
```
Comment (kwalcock) [inline src/main/scala/org/clulab/habitus/printer/MentionInfo.scala:17]: I *think* there might be something off on the right side of this interval.  Isn't it usually math.min(..., _.length)?  length is typically already one beyond the indexable values and adding another would be out of bounds.

Comment (maxaalexeeva) [reply src/main/scala/org/clulab/habitus/printer/MentionInfo.scala:17]: Ah right thanks!

Comment (kwalcock) [reply src/main/scala/org/clulab/habitus/printer/MentionInfo.scala]: Around this line, `mention.document` could just be `document`, but see the next comment.  One could drill down directly to the sentences.

File: src/main/scala/org/clulab/habitus/printer/MentionInfo.scala [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
+    val contextWindow = mention.document.sentences.slice(sentInterval._1, sentInterval._2).map(_.getSentenceText).mkString(" ")
```
Comment (kwalcock) [inline src/main/scala/org/clulab/habitus/printer/MentionInfo.scala:18]: If that sentInterval was not a Tuple2[Int, Int] but rather a Range, then quite a bit of math could be removed and it wouldn't be necessary to program by the numbers with _._1 and _._2, which can be difficult to understand.  Next time I have to do this myself, I will probably write something like
```scala
import org.clulab.struct.Interval

val sentences = mention.document.sentences
val mentionRange = Interval(mention.sentence).indices // for lack of a Range(Int) constructor :-(
val sentRange = Range(mentionRange.start - windowSize, mentionRange.end + windowSize).intersect(sentences.indices)
val contextWindow = sentRange.map(sentences(_).getSentenceText).mkString(" ")
```

**Review by kwalcock — APPROVED** [review]
I learned something about Ranges :-)

File: src/main/scala/org/clulab/habitus/printer/MentionInfo.scala (hunk 2 above)
Comment (maxaalexeeva) [reply src/main/scala/org/clulab/habitus/printer/MentionInfo.scala]: @kwalcock, that looks great! I especially like the intersect there. Updated.

**Comment by maxaalexeeva** [comment]
@kwalcock oops, thanks, forgot to change the document part

**Comment by kwalcock** [comment]
I can be very pedantic.  It's Mihai's turn now.

**Comment by MihaiSurdeanu** [comment]
LGTM!

**Pull request merged**